### DISCUSSION TITLE
Default setting of Main Graph time width doesn't work

### DISCUSSION THREAD

**Discussion by ANDREAPIPPI** [NightscoutFoundation/xDrip#3178] [Bug reports]
Some time ago I noticed that if I set as default the graph main display at 3 hours (or any of the available options), as always did from the start of this feature setting, I obtain the 3 hours setting but if I restart it goes again to a  9 hours width, and that looks to be the default one!

![Graph time width](https://github.com/NightscoutFoundation/xDrip/assets/32884006/db1fb9be-3ba0-49d0-878b-1dfdcf31df71)

I supposed it could be due ro my phone color settings to "dark", that was the reason of strange graph colors and disappeared after setting to "clear", but this problem is not due to this, both "dark" and "clear"! settings give the same problem.
POCO F3- MIUI 14 (Android 13)- Nightly 2023.11.04
Someone else with this problem, or it's a problem of mine?
Thank you
Andrea   :-)

**Comment by Navid200** [comment]
What is this?
![281472409-db1fb9be-3ba0-49d0-878b-1dfdcf31df71](https://github.com/NightscoutFoundation/xDrip/assets/51497406/5ceb8a1b-d713-4519-820c-4c0f0a4fbcb0)

**Reply by ANDREAPIPPI** [reply]
Hi Navid. 
That's heartbeats from Amazfit GTS2 mini through Watchdrip and Zepp...   :-)

**Reply by Navid200** [reply]
Can you disable items one at a time to see which one is causing this?

**Reply by ANDREAPIPPI** [reply]
Sure, I'l do it tomorrow, thank you.

**Comment by ANDREAPIPPI** [comment]
Hi Navid.
I followed your suggestion and modified one at a time all the checked items in display and graph settings, with no variations.
Moreover I forgot to point out that, as shown in the image, the graph extends for many hours ( 6 over 9!), occupying most of the screen with forecasts only.
What else can I check?
Thank you.
Andrea   :-)
Added later: I think I just found it. If I switch OFF the **predictive simulations**, all is as before, the defaukt chosen 3 hours graph width is stable.

![Found](https://github.com/NightscoutFoundation/xDrip/assets/32884006/5a17f189-4df7-4c12-aa86-29ecc0d1c04e)

Hope this helps.
Bye
Andrea  :-
)

**Reply by Navid200** [reply]
This is showing only 3 hours now, isn't it?  
Then, one of the items you have disabled was causing it.  We need to know which.

**Comment by jamorham** [comment]
Is it that the graph auto expands to include predicted values. When there are not predicted values then it uses the selected time window preference?

**Reply by ANDREAPIPPI** [reply]
Yes, that's what happens. 
Is it possible to reduce the time of prevision shown in the graph, e.g. from the  actual 6 hours to about 3-4?
Thank you.
Andrea  :-)

**Reply by Navid200** [reply]
xDrip adds 10 minutes to the current reading and that defines how far in the future the main screen extends to.

But, if there are predictions, their curves need to be shown.  That could extend the time into the future a lot depending on the slope of the predictions.

I have always wanted to modify this slightly so that the predictions are only shown until they drop below the Low value to avoid unreasonable long extensions.
But, have never got to it.
